dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/interceptors/RestrictedInterceptor.java

```diff
@@ -72,12 +72,17 @@ public Object callRestricted(InvocationContext ctx) throws Exception {
     private PermissionException createMissingPermissionException(
             List<MissingPermissions> missingPermissions, Method method, DataverseRequest request) {
         MissingPermissions firstMissing = missingPermissions.get(0);
+        Set<Permission> allMissingPermissions = missingPermissions.stream()
```

Comment:
These missing permissions are connected with certain specific dv objects. I'm not sure If we should gather permissions only.

Also, if I remember correctly, in some places not all permissions are checked, but only first missing is sent to exception. We should discuss if we are going to fix it globally or not.

Reply:
As discussed on a meeting - new task created for this.